src/nodes/typeGuards/compound.ts

```diff
@@ -61,6 +61,16 @@ export function isJsxTagNamePropertyAccess(
 	);
 }
 
+export interface NamedDeclarationWithName extends ts.NamedDeclaration {
+	name: NonNullable<ts.NamedDeclaration["name"]>;
+}
+
+export function isNamedDeclarationWithName(
+	node: ts.Declaration
+): node is NamedDeclarationWithName {
+	return "name" in node;
```

Comment:
I worry about `{ ... name: undefined }` passing this as true. WDYT?

```suggestion
	return "name" in node && !!node.name;
```

Reply:
I don't think it's necessary but it can't hurt.